### PR TITLE
Add MIDA, CoIRLS, distribution plot, and brain example

### LINKED ISSUE #173: Construct an example on brain fMRI classification
Source: https://github.com/kundaMwiza/fMRI-site-adaptation
Attempt: https://github.com/shan3333/Mwiza_Pykale

### PR DESCRIPTION
**Pull request by sz144**
Fixes #173.

### Description
- An example for multi-site brain fMRI classification.
- A toy example of domain adaptation.
- PyTorch version>=1.10.0
- Refactor: `kale.embed.mpca` -> `kale.embed.factorization`
- New features:
    * `kale.embde.factorization.MIDA`
    * `kale.interpret.visualize.distplot_1d`
    * `kale.pipeline.multi_domain_adapter.CoIRLS`

### Status
**Ready**

### Types of changes

- [ ] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] In-line docstrings updated.
- [x] [Source for documentation at `docs`](https://github.com/pykale/pykale/tree/main/docs/source) manually updated for new API.

### CODE REVIEW

**Comment by haipinglu** [comment]
@sz144 I hope to work with you to resolve and merge this PR after merging #284. Share the issues remaining with the team today.

**Comment by XianyuanLiu** [comment]
The [issue](https://github.com/pykale/pykale/runs/6865458874?check_suite_focus=true) seems unrelated to Pyg but Torchvision. Test fails in Torch 1.9.1 and Torchvision 0.10.1, but works in the newer versions, e.g. Torch 1.10.0/1.10.2/1.11.0 and Torchvision 0.11.0/ 0.11.3/0.12.0.

The [zip file](https://drive.google.com/file/d/1U4D23R8u8MJX9KVKb92bZZX-tbpKWtga/view) can be downloaded and unzipped manually, so the file is okay. However, this file seems incomplete when downloaded through `torchvision.datasets.utils.download_file_from_google_drive` in Torchvision 0.10.1, so it raises a unzip fail error. The reason is still unknown. @pz-white Is it the same on your laptop?

Since the newest PyTorch does not support Python 3.6, we may take this opportunity to upgrade our packages, like Python/PyTorch/Torchvision/PyTorch Lightning. The version can be discussed in the meeting. :)

**Comment by haipinglu** [comment]
> 

@XianyuanLiu Many thanks for your input and investigation. 

I suggest the following:
1. @XianyuanLiu Please add this observation to #316 and then comment off the testing of gdrive download API so that this PR can be merged.
2. Then we create a fresh, new branch to upgrade PyKale to remove support for 3.6 and provide support for 3.7 and above, including removing the commented gdrive download testing.

@bobturneruk @sz144 @pz-white What do you think about the above plan?

**Comment by bobturneruk** [comment]
I'll take a look.

**Comment by sz144** [comment]
Let us try if there is any conflicts with newer version of torch and torchvision

**Comment by sz144** [comment]
@bobturneruk PyTorch 1.10.0+Torchvision 0.11.0 works with both Python 3.6 and 3.7, and passed all tests. I think this commit has solved the problems.

File: kale/embed/decomposition.py
Comment (haipinglu) [inline kale/embed/decomposition.py:268]: Is there a test for this new calss? I did not find any. It will be good to have one to maintain our coverage at 93%. Thanks.

File: kale/embed/decomposition.py
Comment (haipinglu) [inline kale/embed/decomposition.py:14]: This comment is about line 1 to 12. Since you think renaming it to decomposition is better, the top docstring should be updated accordingly to specify the reasoning, i.e. what should go here or what is expected here.

File: kale/embed/decomposition.py
Comment (haipinglu) [inline kale/embed/decomposition.py:7]: Consistency of capitalisation for acronyms (MIDA and MPCA differ).

MIDA is a matrix factorization approach, not a matrix factorization, right?

File: kale/embed/decomposition.py
Comment (haipinglu) [inline kale/embed/decomposition.py:6]: Add articles 'a'

File: examples/autism_detection/config.py
Comment (haipinglu) [inline examples/autism_detection/config.py:2]: Have you checked your code/comment? Are you sure this config is about cardiac?

File: examples/autism_detection/config.py
Comment (haipinglu) [inline examples/autism_detection/config.py:18]: Any reference to help users to read further?

File: examples/autism_detection/config.py
Comment (haipinglu) [inline examples/autism_detection/config.py:19]: Any reference (as a comment)?

File: examples/toy_domain_adaptation/main.py
Comment (haipinglu) [inline examples/toy_domain_adaptation/main.py:12]: Is this reusable for problems of the same nature? If yes, then it should go to core API.

File: examples/video_loading/README.md
Comment (haipinglu) [inline examples/video_loading/README.md:62]: Why change this and with two 5s and 6s?

File: kale/embed/decomposition.py
Comment (haipinglu) [inline kale/embed/decomposition.py:6]: As said, factorization should be better than decomposition.
Also, have you updated the docs?

File: setup.cfg
Comment (haipinglu) [inline setup.cfg:9]: The change is no needed, already having # noqa. Better to be local rather than global.

File: kale/pipeline/multi_domain_adapter.py
Comment (haipinglu) [inline kale/pipeline/multi_domain_adapter.py:371]: lb: can you make it more readable (lb stands for pound https://en.wikipedia.org/wiki/Pound_(mass)?

File: kale/embed/factorization.py
Comment (haipinglu) [inline kale/embed/factorization.py:312]: lb: can you make it more readable? 
lb: pound https://en.wikipedia.org/wiki/Pound_(mass)

File: kale/embed/factorization.py
Comment (haipinglu) [inline kale/embed/factorization.py:370]: nl: can you make it more readable?

File: tests/pipeline/test_multi_domain_adapter.py
Comment (haipinglu) [inline tests/pipeline/test_multi_domain_adapter.py:70]: Is coir the same as CoIRLS?

**Review by haipinglu — CHANGES_REQUESTED** [review]
Great job. 
Only some minor issues remain to be fixed (see comments). Once done, it can be merged. Many thanks.

File: kale/embed/factorization.py
Comment (haipinglu) [reply kale/embed/factorization.py:312]: Check all *lb* please

**Review by haipinglu — APPROVED** [review]

**Pull request merged**
